.circleci/config.yml

```diff
@@ -15,6 +15,10 @@ jobs:
           keys:
             - v1-external-dep
             - v1-dep-{{ checksum "Gopkg.lock" }}
+
+      - run: test -z "$(goimports -l -w . | tee /dev/stderr)"
+          name: Ensure goimports has been run
```

Comment:
The `goimports` command should be configured using the tool [pre-commit](https://pre-commit.com/) in a `.pre-commit-config.yaml`. You can then check it, and any other pre-commit hooks, in the CI build via:

```yml
- run: pre-commit install
- run: pre-commit run --all-files
```

Reply:
Got it. Did I choose the right place to insert the `run`, is this where I should add the `pre-commit` hook?

Reply:
Prob better to put it just above `run-go-tests`. BTW, I'm not sure if Circle has `goimports` or `pre-commit` 
installed by default...

Reply:
So it looks pretty common to create your own hooks repo, e.g., for you all it could be `github.com/gruntwork-io/pre-commit`.

Or would you rather use another Github users [go-imports](https://github.com/troian/pre-commit-golang) hook from [pre-commit](https://pre-commit.com/hooks.html) hooks page.

Also, I didn't see many examples of users running this via the CI (travis or CircleCI).

Reply:
Ha, now that you mention it, we actually had a repo with pre-commit hooks already at Gruntwork, but they were in a private repo. I've extracted them into an open source one: https://github.com/gruntwork-io/pre-commit. There's a `go fmt` hook there already. Feel free to use that or add a `goimports` one.

We enforce pre-commit hooks in CI to make sure people are actually running them. The two commands I listed above show how.

Reply:
I added the `goimports` [hook](https://github.com/gruntwork-io/pre-commit/pull/2) to `pre-commit` repo.